app.js

```diff
@@ -100,6 +100,70 @@ function createApp(config) {
     crawlerSecret
   )
 
+  // ===================================================
+  // Log the heap statistics at regular intervals
+  // ===================================================
+  // NOTE: set 'LOG_NODE_HEAPSTATS' env var to 'true' to log heap stats
+  // NOTE: set 'LOG_NODE_HEAPSTATS_INTERVAL_MS' env var to '<time_in_milliseconds>' for logging interval
+  const shouldLogHeapstats = config.heapstats.logHeapstats
+    ? config.heapstats.logHeapstats.toLowerCase() === 'true'
+    : false
+
+  if (shouldLogHeapstats) {
+    const v8 = require('v8')
+
+    const addCommas = num => Number(num).toLocaleString()
+    const isNumeric = num => !isNaN(Number(num))
+
+    // Set the heapstats logging interval
+    const maybeInterval = config.heapstats.logInverval
+    const heapStatsInverval = maybeInterval && isNumeric(maybeInterval) ? maybeInterval : 30000
+
+    // Function to log the heap space statistics
+    const logHeapSpaceStats = () => {
+      // Get the current timestamp
+      const currentTimestamp = new Date().toISOString()
+
+      // Get the heap space statistics
+      const heapSpaceStats = v8.getHeapSpaceStatistics()
+
+      heapSpaceStats.forEach(space => {
+        const heapStatsMessage =
+          `[${currentTimestamp}] Heap Space Statistics: ` +
+          `Space Name: '${space.space_name}', ` +
+          `Space Size: '${addCommas(space.space_size)}' bytes, ` +
+          `Space Used Size: '${addCommas(space.space_used_size)}' bytes, ` +
+          `Space Available Size: '${addCommas(space.space_available_size)}' bytes, ` +
+          `Physical Space Size: '${addCommas(space.physical_space_size)}' bytes` +
+          '\n--------------------------'
+
+        logger.info(heapStatsMessage)
+      })
+
+      // Get the heap statistics
+      const heapStats = v8.getHeapStatistics()
+
+      const heapStatsMessage =
+        `[${currentTimestamp}] Heap Statistics: ` +
+        `Total Heap Size: '${addCommas(heapStats.total_heap_size)}' bytes, ` +
+        `Total Heap Used Size: '${addCommas(heapStats.total_heap_size)}' bytes, ` +
```

Comment:
duplicate of line 148?  total_heap_size_executable instead?

Reply:
Yep!  Thank you for catching that!  Updated the duplicate line to be "total_heap_size_executable"